### ISSUE TITLE
[yugen] ssl certificate for custom domain is not being updated

### ISSUE THREAD

**Issue by webratz** [glomex/gcdt#367]
@Dgadavin reported issues with SSL certificates in API gateway not being updated.
While checking the code it actually appears that the API gateway custom domain config is not being updated once that domain exists. Therefore the SSL certificate is never updated.

See: https://github.com/glomex/gcdt-yugen/blob/06ddd8fa857dc5d5eb58166721931e4007979632/gcdt/yugen_core.py#L210

While working on this, it might also be worth adding support for ACM instead of the current direct upload.

**Comment by webratz** [comment]
Had a look on the botocore docs, and the current way seems to be deprecated. (http://botocore.readthedocs.io/en/latest/reference/services/apigateway.html#APIGateway.Client.create_domain_name )
So the solution should be using ACM anyway.

**Comment by markfink** [comment]
I think it is best to remove the deprecated certificate use in create_custom_domain and replace it with the suggested 'certificateArn' plus to add the update feature. But this might break somebodys code. If this is not super urgent I will bring this topic up in the next gcdt grooming so we can communicate the changes beforehand.

**Comment by markfink** [comment]
I think certificate handling in yugen works as designed so I removed the 'bug' label. the update feature was never implemented by data_platform.

**Comment by Dgadavin** [comment]
@markfink  It's urgent for Data Platform. They cant do normal deployment because `yugen` alway use old SSL. Would be good to have quick hotfix.

**Comment by markfink** [comment]
I released gcdt version 0.1.437 that contains the hotfix
docs here: http://gcdt.readthedocs.io/en/latest/50_yugen.html#create-custom-domain

you need to both update gcdt and gcdt-lookups
probably easiest is to use the following command:
```
$ pip install -U -r requirements_gcdt.txt
```

your config needs to contain both certificateName and a lookup for the certificateArn like this:
``` 
   "customDomain": {
      "basePath": "",
      "certificateName": "wildcard.glomex.com-2017-3-2",
      "certificateArn": "lookup:acm:*.infra.glomex.cloud",
      "domainName": "unittest-gcdt-sample-api-dev-eu-west-1.dev.mes.glomex.cloud",
      "hostedDomainZoneId": "lookup:stack:infra-dev:internalDomainHostedZoneID",
      "route53Record": "unittest-gcdt-sample-api-dev-eu-west-1.dev.infra.glomex.cloud"
    }
```

**Comment by markfink** [comment]
feedback from @Dgadavin was:
```
Everything works good for UM sqaud. DP is still in progress, but I assume everything will be Ok
Have feedback from DP. Everything work. Big thanks @mark
```
